File: app/src/main/java/com/codingblocks/restapiretrofitjson/activities/UsersActivity.java
package com.codingblocks.restapiretrofitjson.activities;

import android.content.Intent;
import android.graphics.drawable.ColorDrawable;
import android.os.Build;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.Button;

import com.codingblocks.restapiretrofitjson.R;
import com.codingblocks.restapiretrofitjson.adapters.UserAdapter;
import com.codingblocks.restapiretrofitjson.api.API;
import com.codingblocks.restapiretrofitjson.interfaces.OnItemClickListener;
import com.codingblocks.restapiretrofitjson.models.User;

import java.util.ArrayList;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class UsersActivity extends AppCompatActivity {

    public static final String TAG = "USERS";

    RecyclerView rvUserList;
    UserAdapter userAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_users);

        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP)
            getWindow().setStatusBarColor(getResources().getColor(R.color.colorPrimary));
        getSupportActionBar().setBackgroundDrawable(new ColorDrawable(getResources().getColor(R.color.colorPrimary)));

        rvUserList = (RecyclerView) findViewById(R.id.rvUserList);
        rvUserList.setLayoutManager(new LinearLayoutManager(this));
        userAdapter = new UserAdapter(this, new ArrayList<User>());
        rvUserList.setAdapter(userAdapter);
        userAdapter.setOnItemClickListener(new OnItemClickListener() {
            @Override
            public void onItemClick(int itemId, View view) {
                Intent intent = null;

                switch (view.getId()) {
                    case R.id.btnUserTodos:
                        intent = new Intent(UsersActivity.this,
                                TodosActivity.class);
                        break;
                    case R.id.btnUserPosts:
                        intent = new Intent(UsersActivity.this,
                                PostsActivity.class);
                        break;
                    case R.id.btnUserAlbum:
                        intent = new Intent(UsersActivity.this,
                                AlbumsActivity.class);
                        break;
                }

                intent.putExtra("userId", itemId);
                startActivity(intent);
            }
        });

        ((Button) findViewById(R.id.btnAllPosts)).setOnClickListener(
                new View.OnClickListener() {
                    @Override
                    public void onClick(View view) {
                        startActivity(new Intent(
                                UsersActivity.this,
                                PostsActivity.class
                        ));
                    }
                }
        );


        API.getInstance().getUsersAPI().getUsers().enqueue(new Callback<ArrayList<User>>() {
            @Override
            public void onResponse(Call<ArrayList<User>> call,
                                   Response<ArrayList<User>> response) {
                userAdapter.updateUsers(response.body());
            }

            @Override
            public void onFailure(Call<ArrayList<User>> call, Throwable t) {

            }
        });

    }


}

File: app/src/main/java/com/codingblocks/restapiretrofitjson/activities/UsersActivity.kt
package com.codingblocks.restapiretrofitjson.activities

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.view.View
import android.widget.Button
import com.codingblocks.restapiretrofitjson.R
import com.codingblocks.restapiretrofitjson.adapters.UserAdapter
import com.codingblocks.restapiretrofitjson.api.API
import com.codingblocks.restapiretrofitjson.models.User
import kotlinx.android.synthetic.main.activity_users.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import java.util.*

class UsersActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_users)

        rvUserList.layoutManager = LinearLayoutManager(this)
        val userAdapter = UserAdapter(this, ArrayList<User>())
        rvUserList.adapter = userAdapter
        pbProgress.visibility = View.VISIBLE
        supportActionBar!!.title = "Users"

        (findViewById(R.id.btnAllPosts) as Button).setOnClickListener {
            startActivity(Intent(
                    this@UsersActivity,
                    PostsActivity::class.java
            ))
        }


        API.getInstance().usersAPI.users.enqueue(object : Callback<ArrayList<User>> {
            override fun onResponse(call: Call<ArrayList<User>>,
                                    response: Response<ArrayList<User>>) {
                userAdapter.updateUsers(response.body()!!, pbProgress)
            }

            override fun onFailure(call: Call<ArrayList<User>>, t: Throwable) {

            }
        })

    }

    companion object {

        val TAG = "USERS"
    }


}
